File: Lab 17 A-B Person App/A/PersonApp/src/me/chris/PersonArray.java
package me.chris;

public class PersonArray {
	private Person[] people;
	private int numPeople;
	
	public PersonArray(int personCount) {
		this.people = new Person[personCount];
		this.numPeople = personCount;
	}
	
	public void insert(String firstName, String lastName, int age) {
		for(int i=0; i<this.numPeople; i++) {
			if (this.people[i] == null) {
				this.people[i] = new Person(firstName, lastName, age);
				break;
			}
		}
	}
	
	public void delete(String lastName) {
		Person[] tmp = new Person[this.numPeople-1];
		int count = 0;
		for(int i = 0; i < this.numPeople; i++) {
			if (!this.people[i].getLastName().equals(lastName)) {
				tmp[count] = this.people[count];
				count++;
			}
		}
		this.numPeople -= 1;
		this.people = tmp;
	}
	
	public Person find(String lastName) {
		for(int i = 0; i < this.numPeople; i++) {
			if (this.people[i].getLastName().equals(lastName)) {
				return this.people[i];
			}
		}
		return null;
	}
	
	public void DisplayAll() {
		for(int i = 0; i< this.numPeople; i++) {
			System.out.println(this.people[i]);
		}
	}
}

File: Lab 17 A-B Person App/B/PersonAPp/PersonArray.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PersonAPp
{
    class PersonArray
    {
        private Person[] people;
        private int numPeople;
        public PersonArray(int personCount)
        {
            this.numPeople = personCount;
            people = new Person[personCount];
        }

        public void Insert(String firstName, String lastName, int age)
        {
            for(var i=0; i<numPeople; i++)
            {
                if (people[i] == null)
                {
                    people[i] = new Person(firstName, lastName, age);
                    break;
                }
            }
        }

        public void Delete(String lastName)
        {
            Person[] tmp = new Person[this.numPeople-1];
            var count = 0;
            for(var i = 0; i < numPeople; i++)
            {
                if (!people[i].lastName.Equals(lastName))
                {
                    tmp[count] = people[count];
                    count++;
                }
            }
            numPeople -= 1;
            people = tmp;
        }

        public Person Find(String lastName)
        {
            foreach (Person p in people)
            {
                if (p.lastName.Equals(lastName))
                {
                    return p;
                }
            }
            return null;
        }

        public void DisplayAll()
        {
            for(var i = 0; i< numPeople; i++)
            {
                Console.WriteLine(people[i]);
            }
        }
    }
}
